src/rollup/BridgeCreator.sol

```diff
@@ -84,8 +86,18 @@ contract BridgeCreator is Ownable {
         address nativeToken,
         ISequencerInbox.MaxTimeVariation calldata maxTimeVariation
     ) external returns (BridgeContracts memory) {
+        // make sure the rollup exists
+        // otherwise the rollup creator could be DOSed by frontrunning
+        if (rollup.code.length == 0) {
+            revert RollupCodeEmpty();
+        }
+
+        // use create2 salt to ensure deterministic addresses
+        bytes32 create2Salt = keccak256(abi.encode(adminProxy, rollup, nativeToken, maxTimeVariation));
```

Comment:
technically `adminProxy` isn't required in the salt since it's a constructor arg. can't hurt to include it though IMO

Reply:
using the entire msg.data